src/core/providers/sensorthings/qgssensorthingsshareddata.cpp

```diff
@@ -62,8 +62,15 @@ QgsSensorThingsSharedData::QgsSensorThingsSharedData( const QString &uri )
     {
       mGeometryType = Qgis::WkbType::MultiPolygonZ;
     }
-    // geometry is always GeoJSON spec (for now, at least), so CRS will always be WGS84
-    mSourceCRS = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4326" ) );
+    else if ( !uriParts.contains( QStringLiteral( "geometryType" ) ) )
```

Comment:
@nyalldawson , I feel the logic would be easier to read if you did something like:

```
if ( uriParts.contains( "geometryType" )
{
  const QString geometryType = uriParts.value( QStringLiteral( "geometryType" ) ).toString();
  if ( geometryType.compare( QLatin1String( "point" ), Qt::CaseInsensitive ) == 0 )
  {
    mGeometryType = Qgis::WkbType::PointZ;
  }
  else if
  {
  ...
  mSourceCRS = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4326" ) );
}
else
{
  mGeometryType = Qgis::WkbType::NoGeometry;
}
```

You'd also then skip the pointless 4 string compare.

Reply:
Indeed, I've done that now!